core/coredatainit.go

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+type CoredataCredentials struct {
+	Name     string `json:"name,omitempty"`
+	Password string `json:"password,omitempty"`
+}
+
+func coredataCredentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
+
+	return credentialInStore(config, secretBaseURL, credPath, c)
+}
+
+func initCoredataCredentials(config *tomlConfig, secretBaseURL string, c *http.Client) error {
+
+	password, _ := createCredential()
+
+	body := &CoredataCredentials{
+		Name:     Coredata,
+		Password: password,
+	}
+
```

Comment:
How does the above credential relate to the username/password provided to the MongoCredentials struct below? 
```
coreadatapasswd, _ := createCredential()

body := &MongoCredentials{
...
    CoredataUser:        "coredata",
    CoredataPasswd:      coreadatapasswd,
...
}
```
Are the `CoredataCredentials` used for database access or something else? Because presumably the password value generated in the two places will be different.